core/handlers.py

```diff
@@ -65,23 +67,30 @@ def sv_handler_undo_pre(scene):
 
 @persistent
 def sv_handler_undo_post(scene):
-    print('called undo post')
+
     # this function appears to be hoisted into an environment that does not have the same locals()
     # hence this dict must be imported. (jan 2019)
 
     from sverchok.core import undo_handler_node_count
 
     num_to_test_against = 0
+    links_changed = False
     for ng in sverchok_trees():
         num_to_test_against += len(ng.nodes)
-
+        tree_id = ng.tree_id
+        if tree_id in sv_links_cache:
+            links_changed = sv_links_cache[tree_id] != bl_links_to_sv_links(ng)
+            if links_changed:
+                break
```

Comment:
You are doing this test in many places in the code. You should create method for this test. If you have to change test it will be possible to do in one place.

Reply:
this line is only one time in the code, it is similar in another spot but different because i need to store the links first in a variable

Reply:
It does not matter is it the line strictly the same or not. If the test of changing of links will change (most likely it can happen) you have to scan all code in catching such places. And this task will much harder after a few months than it looks now. It is much easier to have one common method which can do such operation.

Reply:
This is one call:
```
if tree_id in sv_links_cache:
   links_changed = sv_links_cache[tree_id] != bl_links_to_sv_links(ng)
   if links_changed:
       break
```
And this is the other one:
```
new_sv_links = bl_links_to_sv_links(node_tree)
before_sv_links = sv_links_cache[tree_id]
links_has_changed = before_sv_links != new_sv_links
if links_has_changed:
   some_var = some_func(new_links)
```

As you can see they are different because in the second case i need to store the links first.
If it changes in the future, the future coder will have also to deal with opening, undoing, updating, reloading and sub-trees. That line would be the less of his problems.

Reply:
I see nothing what can prevent to move this code to separate method.
If the goal is only to change the logic link changes detection there is no need in investigation all code of this PR. 
For example some other events can be used for changes detection. It will let avoid expensive comparison between old and new links. New logic can be more tricky and consist from tens of strings. It will be hard to search all places where new code should be pasted. And most likely new code will be separate function any case. And it will be much simpler if new code just replace code of one method.